Essentials/src/main/java/com/earth2me/essentials/commands/Commandhome.java

```diff
@@ -73,10 +73,14 @@ public void run(final Server server, final User user, final String commandLabel,
                 final List<String> homes = finalPlayer.getHomes();
                 if (homes.isEmpty() && finalPlayer.equals(user)) {
                     if (ess.getSettings().isSpawnIfNoHome()) {
-                        final UserTeleportHomeEvent event = new UserTeleportHomeEvent(user, null, bed != null ? bed : finalPlayer.getWorld().getSpawnLocation(), bed != null ? UserTeleportHomeEvent.HomeType.BED : UserTeleportHomeEvent.HomeType.SPAWN);
-                        server.getPluginManager().callEvent(event);
-                        if (!event.isCancelled()) {
-                            user.getAsyncTeleport().respawn(charge, TeleportCause.COMMAND, getNewExceptionFuture(user.getSource(), commandLabel));
+                        if(bed == null || finalPlayer.isAuthorized("essentials.home.bed")){
+                            final UserTeleportHomeEvent event = new UserTeleportHomeEvent(user, null, bed != null ? bed : finalPlayer.getWorld().getSpawnLocation(), bed != null ? UserTeleportHomeEvent.HomeType.BED : UserTeleportHomeEvent.HomeType.SPAWN);
+                            server.getPluginManager().callEvent(event);
+                            if (!event.isCancelled()) {
+                                user.getAsyncTeleport().respawn(charge, TeleportCause.COMMAND, getNewExceptionFuture(user.getSource(), commandLabel));
+                            }
+                        }else {
```

Comment:
Please fix the indentation here and in other places in the code to match the code style